presto-orc/src/main/java/com/facebook/presto/orc/stream/BooleanInputStream.java

```diff
@@ -132,13 +133,68 @@ public int getSetBits(int batchSize, boolean[] vector)
             throws IOException
     {
         int count = 0;
-        for (int i = 0; i < batchSize; i++) {
+        int bitsInFirstByte = Math.min(batchSize, bitsInData);
```

Comment:
I haven't seen a reply to this comment:

---
this change doesn't really belong to this PR; would you submit it to master?